python/pyspark/ml/torch/deepspeed/tests/test_deepspeed_distributor.py

```diff
@@ -0,0 +1,162 @@
+            self.assertEqual(local_cmd_with_args,LOCAL_CMD_ARGS_EXPECTED)
+
+    def test_create_torchrun_command_distributed(self):
+        DEEPSPEED_CONF = "path/to/deepspeed"
+        TRAIN_FILE_PATH = "path/to/exec"
+        NUM_PROCS = 10
+        input_params = {}
+        input_params["local_mode"] = True
+        input_params['num_processes'] = NUM_PROCS
+        input_params["deepspeed_config"] = DEEPSPEED_CONF
+        # distributed training environment
+        distributed_master_address, distributed_master_port, distributed_rank = self._get_env_variables_distributed()
+        distributed_torchrun_args = [
+                f"--nnodes={NUM_PROCS}",
+                f"--node_rank={distributed_rank}",
+                f"--rdzv_endpoint={distributed_master_address}:{distributed_master_port}",
+                "--rdzv_id=0",
+            ]
+        DISTRIBUTED_CMD_NO_ARGS_EXPECTED = [sys.executable,
+                                            "-m",
+                                            "torch.distributed.run",
+                                            *distributed_torchrun_args,
+                                            "--nproc_per_node=1",
+                                            TRAIN_FILE_PATH,
+                                            "-deepspeed",
+                                            "--deepspeed_config",
+                                            DEEPSPEED_CONF
+                                            ]
+        # test distributed training without arguments
+        input_params["local_mode"] = False
+        distributed_command = DeepspeedTorchDistributor._create_torchrun_command(input_params, TRAIN_FILE_PATH)
+        self.assertEqual(DISTRIBUTED_CMD_NO_ARGS_EXPECTED,distributed_command)
+        # test distributed training with random arguments
+        distributed_extra_args = ["-args1", "--args2"]
+        DISTRIBUTED_CMD_ARGS_EXPECTED = [sys.executable,
+                                            "-m",
+                                            "torch.distributed.run",
+                                            *distributed_torchrun_args,
+                                            "--nproc_per_node=1",
+                                            TRAIN_FILE_PATH,
+                                            *distributed_extra_args,
+                                            "-deepspeed",
+                                            "--deepspeed_config",
+                                            DEEPSPEED_CONF
+                                            ]
+        print("The distributed training command: ", DISTRIBUTED_CMD_ARGS_EXPECTED)
+        distributed_command_with_args = DeepspeedTorchDistributor._create_torchrun_command(input_params, TRAIN_FILE_PATH,*distributed_extra_args)
+        self.assertEqual(DISTRIBUTED_CMD_ARGS_EXPECTED,distributed_command_with_args)
```

Comment:
It's inconsistent that there are sub-tests above for args vs no args but not here.

Reply:
done